src/components/menu/MenuState.tsx

```diff
@@ -0,0 +1,65 @@
+// (C) 2007-2018 GoodData Corporation
+import * as React from 'react';
+
+export interface IMenuStateConfig {
+    opened?: boolean;
+    defaultOpened?: boolean;
+    onOpenedChange?: (opened: boolean) => void;
+}
+
+export interface IMenuStateProps extends IMenuStateConfig {
+    children: (
+        props: {
+            opened: boolean;
+            onOpenedChange: (opened: boolean) => void;
+        }
+    ) => React.ReactNode;
+}
+
+export interface IMenuStateState {
+    opened: boolean;
+}
+
+export default class MenuState extends React.Component<IMenuStateProps, IMenuStateState> {
+    public static defaultProps = {
+        defaultOpened: false
+    };
+
+    constructor(props: IMenuStateProps) {
+        super(props);
+
+        this.state = {
```

Comment:
Consider updating state even for controlled mode to make sure the state is always correct when switching between controlled and uncontrolled mode